### ISSUE TITLE
[BUG] After upgrading from v0.8.1 to v1.0.0 the node and disk tags are removed.

### ISSUE THREAD

**Issue by harutyundermenjyan** [longhorn/longhorn#1440]
**Describe the bug**
Installed as Rancher catalog app v0.8.1 then tried to upgrade to v1.0.0 as it described [here](https://longhorn.io/docs/1.0.0/deploy/upgrade/longhorn-manager/#upgrading-from-v081-to-v100) and after the upgrade the node and disk tags were removed.

**To Reproduce**
1. Go to 'Node'. 
2. Click on 'Edit Node and Disks and Add some disk and/or node Tags'
3. Perform 'upgrade from v0.8.1 to v1.0.0  as it described [here](https://longhorn.io/docs/1.0.0/deploy/upgrade/longhorn-manager/#upgrading-from-v081-to-v100)'
4. After the upgrade, the node and disk tags are removed.

**Environment:**
 - Longhorn version: v1.0.0
 - Kubernetes version:  v1.17.4
 - Node OS type and version: Ubuntu16.04 and 18.04

**Comment by khushboo-rancher** [comment]
I tried upgrade from v0.8.1 to 1.0.0 but couldn't reproduce the issue.

Steps followed:
1. Set up rancher with 4 nodes - ubuntu 18.04 (1 etc/control plane, 3 workers)
2. Install longhorn app.
3. Add node/disk tags for node and disk in longhorn.
4. Upgrade longhorn app to v1.0.0 (By using kubectl command and by app upgrade)
5 Check the added tags for the nodes.

**Comment by harutyundermenjyan** [comment]
Thanks for checking, can't say why it happened in my case. If you checked and all works well please consider the issue solved. I'll check for the reason internally because I'm not alone on that cluster. Thanks again.

**Comment by boknowswiki** [comment]
@harutyundermenjyan 
Do you mind to share more information on the steps and your cluster information? Please feel free to insert your answer or something you think is different to the steps I described below.

1. Use Rancher catalog to install Longhorn v0.8.1 with default settings(didn't change anything and assuming you have a 3 nodes cluster with rancher, please let me know if you have something different).
2. Add node tag and disk tags via Lognhorn UI. E.g, adding `dev` to all the nodes, and adding `default` disk tag to the default disks, and `extra` disk tag to extra longhorn disks. And create a longhorn volume with node tag `dev`, disk tag `default`. (Do you have something similar or you have something different?)
3. Create a k8s pod named `longhorn-simple-pod` using [PVC yaml file](https://raw.githubusercontent.com/longhorn/longhorn/master/examples/simple_pvc.yaml) and  [pod yaml file](https://raw.githubusercontent.com/longhorn/longhorn/master/examples/simple_pod.yaml) create a longhorn volume.
3. Follow [preparing the upgrade](https://longhorn.io/docs/1.0.0/deploy/upgrade/longhorn-manager/#preparing-for-the-upgrade).
    a. Longhorn `StorageClass` parameters are the same as default.
    b. Delete the `longhorn-simple-pod` pod. (If it's a replica set, need to set the replica to 0 or delete it). But the PVC is still there in detached state.
4. Use `kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/longhorn.yaml` to deploy Longhorn v1.0.0. ( Please let me know if you were using Helm).
5. Select all the volumes using batch selection. Click the batch operation button `Upgrade Engine`

And would like to double confirm with you about the step 4 you mentioned, you mean the tags for the node and disk(node tag and disk tag) are remove, right?

Thanks.

**Comment by harutyundermenjyan** [comment]
Yes, the steps you provided are very similar I did, and you right about step 4.
I'm not refuting that something I missed/confused during the upgrade because after you check this, I've repeated the steps and all went well and all tags are preserved as well. The problem is that steps I've given not performed immediately one after the other but there were some gaps between them so I assume something happened during the mentioned gaps(e.g. the tags are deleted manually by someone other.. I'll check it internally).
Thank you and sorry.
P.S.
Great product.
